index.ts

```diff
@@ -12,6 +12,22 @@ type Attributes = JSX.IntrinsicElements['div'];
 type DocumentFragmentConstructor = typeof DocumentFragment;
 type ElementFunction = () => HTMLElement | SVGElement;
 
+declare global {
+	namespace JSX {
+		interface Element extends HTMLElement, SVGElement, DocumentFragment {
+			addEventListener: HTMLElement['addEventListener'];
+			removeEventListener: HTMLElement['removeEventListener'];
+			className: HTMLElement['className'];
```

Comment:
Both `HTMLElement` and `SVGElement` have the listeners, why specify them here?
```suggestion
```

Reply:
not specifying them will lead to erros like the one below, moving the declaration to `global.d.ts` will result in `JSX.Element` not being included in the compiled `index.d.ts` file and other errors, so redeclaring the properties seems to be necessary.
```
Interface 'Element' cannot simultaneously extend types 'HTMLElement' and 'DocumentFragment'.
  Named property 'addEventListener' of types 'HTMLElement' and 'DocumentFragment' are not identical.ts(2320)
```

Reply:
TypeScript’s declarations confuse the hell out of me. Why does this work without issue on Refined GitHub’s global.d.ts while it doesn't here? Is it because RGH also includes `@types/react`? (also suggested)

Reply:
i think its because the `RGH/globals.d.ts` file is in the project folder so the language server (and compiler) knows about the types, in `dom-chef` the `dom-chef/global.d.ts` file doesn't get compiled because [typescript doesn't compile `d.ts` files](https://stackoverflow.com/a/49814709/11202842).

Reply:
including `@types/react` in the dep section and not in the devDep section might be a good idea since you have to add if you use typescript, and in plain js the `d.ts` files wont be used so it should not impact plain js projects at all.